core/tests/AzureMcp.Core.UnitTests/Areas/Server/Commands/ToolLoading/CommandFactoryToolLoaderTests.cs

```diff
@@ -290,6 +290,61 @@ public async Task CallToolHandler_WithUnknownTool_ReturnsError()
         Assert.Contains("Could not find command: non-existent-tool", textContent.Text);
     }
 
+    [Fact]
+    public async Task GetsToolsWithRawMcpInputOption()
```

Comment:
I still do not understand why you need this special case within the generic tool loaders.  We need to figure out a different way to handle your input schema.